src/components/RequestAirdrop.vue

```diff
@@ -0,0 +1,43 @@
+<script setup>
+import store from '../vuex';
+import { useWallet } from 'solana-wallets-vue'
+import { Connection, clusterApiUrl, LAMPORTS_PER_SOL } from '@solana/web3.js';
+
+const { publicKey, connected } = useWallet();
+
+function setBalance (balance) {
+    store.commit('updateBalance', { balance });

```

Comment:
Case in point regarding my previous comment. Here you want to use the `setBalance` commit because `updateBalance` increments the balance. So here you're effectively doubling the balance of the user because you're passing through the entire balance.